### ISSUE TITLE
Better file header

### ISSUE THREAD

**Issue by cking** [DevTeam/Immutype#2]
I am using `<TreatWarningsAsErrors>` in my code.

Sadly, using it on your code throws generates a lot of warnings (und thus errors in my case)

Could you add the following Code in your generated file headers:

```csharp
// <auto-generated />
#pragma warning disable CS1591
```

as an alternative to the `<auto-generated />` comment you could decorate each generated method with the following (replacing Immutype and/or 1.0.0.0 with apropriate data:
```csharp
[global::System.CodeDom.Compiler.GeneratedCodeAttribute("Immutype", "1.0.0.0")]
```

**Comment by cking** [comment]
You could add XML-Docs to all generated methods as well (actually preferred) to not need to disable CS1591, but that is way more work

**Comment by NikolayPianikov** [comment]
> You could add XML-Docs to all generated methods as well (actually preferred) to not need to disable CS1591, but that is way more work

This code generator automatically adds comments to methods if there are comments to parameters. For example, [this code](https://github.com/DevTeam/csharp-interactive/blob/042a52bbefece4c124ddd2db7d15d9e14b71556b/CSharpInteractive.HostApi/DotNetBuild.cs#L21) will contain comments for all methods created by the generator.

**Comment by cking** [comment]
I try that, but there are no doc comments for the target attribute though

**Comment by cking** [comment]
Oh, it doesn't get the comment from the xml docs? That's a shame

No wonder it didn't work for me.

**Comment by NikolayPianikov** [comment]
> Oh, it doesn't get the comment from the xml docs? That's a shame
> 
> No wonder it didn't work for me.

Added task #4

**Comment by NikolayPianikov** [comment]
Fixed in 1.0.16
